### PR TITLE
feat: link creation form

### LINKED ISSUE #18: Link creation form
Capability to create links is supported via GraphQL if authenticated.

In order to support link creation from UI a component which holds a form with relevant values is needed.

Currently to create a link we just need:

- URL: The URL where the user must be redirected
- Custom Hash: A custom hash to refer to the link

### PR DESCRIPTION
**Pull request by Michael-Liendo**

### CODE REVIEW

File: src/routes/(app)/components/CreateLink/CreateLink.svelte
Comment (LeoBorai) [inline src/routes/(app)/components/CreateLink/CreateLink.svelte]:
```suggestion
  const urqlClient = getContextClient();
  const { handleSubmit, values, errors, isSubmitting } = newForm({
```

File: src/routes/(app)/components/CreateLink/CreateLink.svelte
Comment (LeoBorai) [inline src/routes/(app)/components/CreateLink/CreateLink.svelte:40]: I think we can improve this a bit later by using the Authorization Extension from URQL

File: src/routes/(app)/components/CreateLink/CreateLink.svelte
Comment (LeoBorai) [inline src/routes/(app)/components/CreateLink/CreateLink.svelte:52]: Well thought!

File: src/routes/(app)/components/CreateLink/CreateLink.svelte
Comment (LeoBorai) [inline src/routes/(app)/components/CreateLink/CreateLink.svelte]:
```suggestion
          label="URL"
```

Perhaps you are trying to achieve a "Required" field approach? I think we could have a prop for that which renders certain UI if specified.

File: src/routes/(app)/components/CreateLink/CreateLink.svelte
Comment (LeoBorai) [inline src/routes/(app)/components/CreateLink/CreateLink.svelte]:
```suggestion
        >Create Link</Button
```

File: src/routes/(app)/components/CreateLink/graphql/LinkCreate.gql
Comment (LeoBorai) [inline src/routes/(app)/components/CreateLink/graphql/LinkCreate.gql]: I recommend using a fragment for cases like this, so we have a type generated for validations

**Review by LeoBorai — COMMENTED** [review]
Looking good so far! I suggested some changes where we still have space for improvements.

It would also be great to have some tests on form and validation perhaps?

Here is a very good article on this field: https://sveltesociety.dev/recipes/testing-and-debugging/unit-testing-svelte-component

File: src/routes/(app)/components/CreateLink/graphql/LinkCreate.gql
Comment (Michael-Liendo) [reply src/routes/(app)/components/CreateLink/graphql/LinkCreate.gql]: You can explain more? I think that I used this fragment

Comment (Michael-Liendo) [reply src/routes/(app)/components/CreateLink/graphql/LinkCreate.gql]: I understand, sure!

File: src/routes/(app)/components/CreateLink/CreateLink.svelte
Comment (Michael-Liendo) [reply src/routes/(app)/components/CreateLink/CreateLink.svelte:40]: How we can use the authorization extension?

Comment (LeoBorai) [reply src/routes/(app)/components/CreateLink/CreateLink.svelte:40]: The Authentication Exchange is part of the URQL ecosystem, you can read more about it here:

https://formidable.com/open-source/urql/docs/api/auth-exchange/

But lets address that in a separate PR, after merging this.

**Review by LeoBorai — APPROVED** [review]
Nice work! I think we are good to go with this PR, if you feel like moving from draft to open!

**Comment by LeoBorai** [comment]
@Michael-Liendo please remember to run: `pnpm run format`, and turn the PR "open" if you consider we are ready here!

**Pull request merged**
